torch/csrc/jit/codegen/cuda/test/test_gpu_match_frontend.cpp

```diff
@@ -0,0 +1,365 @@
+#if defined(USE_CUDA)
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <torch/csrc/jit/codegen/cuda/arith.h>
+#include <torch/csrc/jit/codegen/cuda/executor.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_builder.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/all_schedulers.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/utils.h>
+
+#include <test/cpp/jit/test_utils.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_gpu_validator.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_utils.h>
+
+#include <torch/torch.h>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+using namespace torch::jit::fuser::cuda;
+
+//! A very simple test computing sum(x * 3.0, dim=-1) for 2D inputs
+TEST_F(NVFuserTest, FusionFrontendSuperBasic_CUDA) {
+  int y = 4, z = 8;
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::Tensor t0 = at::randn({y, z}, options);
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2); // {i0, i1}
+  auto c0 = IrBuilder::create<Double>(3.0);
+
+  fusion.addInput(tv0);
+
+  auto tv1 = mul(tv0, c0); // {i0, i1}
+  auto tv2 = sum(tv1, {-1}, false, DataType::Float); // {i0, r1}
+
+  fusion.addOutput(tv2);
+
+  // {i0, r1}
+  tv2->split(
+      1,
+      NamedScalar::getParallelDim(ParallelType::TIDx)); // {i0, r1 / TIDx, TIDx}
+  tv2->axis(0)->parallelize(ParallelType::BIDx);
+  tv2->axis(-1)->parallelize(ParallelType::TIDx);
+  auto tv3 = tv2->rFactor({1});
+
+  // propagate the mapping to other tensors
+  TransformPropagatorWithCheck propagator(tv3);
+  MaxRootDomainInfoSpanningTree(tv3).traverse(&propagator);
+  scheduler_utils::parallelizeAllLike(tv3, {tv0, tv1});
+
+  tv0->computeAt(tv3, 1, ComputeAtMode::MostInlined);
```

Comment:
I think we use `inlineMost()` in our schedulers.

Reply:
Thanks. I didn't know about `inlineMost()` but have switched to using it.